master/cluster.go

```diff
@@ -624,10 +624,18 @@ func (c *Cluster) checkLeaderAddr() {
 
 func (c *Cluster) checkDataNodeHeartbeat() {
 	tasks := make([]*proto.AdminTask, 0)
+	c.volMutex.RLock()
+	defer c.volMutex.RUnlock()
 	c.dataNodes.Range(func(addr, dataNode interface{}) bool {
 		node := dataNode.(*DataNode)
 		node.checkLiveness()
 		task := node.createHeartbeatTask(c.masterAddr(), c.diskQosEnable)
+		hbReq := task.Request.(*proto.HeartBeatRequest)
```

Comment:
insert volMutex lock here, to reduce lock range.

Reply:
Yes, it will be moved to there.